app/models/manageiq/providers/cloud_manager/virtual_template.rb

```diff
@@ -0,0 +1,14 @@
+class ManageIQ::Providers::CloudManager::VirtualTemplate < ::MiqTemplate
+  validate :single_template, :on => :create
+  default_value_for :cloud, true
+
+  TYPES = {
+    :amazon => 'ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate'
+  }.freeze
+
+  def single_template
+    multiple = !type.constantize.where(:type => type).empty?
+    errors.add(:virtual_template, 'may only have one per type') if multiple
```

Comment:
They should definitely be translated as they usually appear in flash messages.

Separately though, on this line, I'm curious why a method called `single_template` would have a side effect of changing the errors.  Since it's returning a boolean I think it should named like a predicate, so `single_template?`.  However, with the errors changing perhaps it should be broken two methods with the second being something like `validate_single_template`.

Something like this:

```ruby
def single_template?
  type.constantize.where(:type => type).empty?
end

def validate_single_template
  errors.add(:virtual_template, 'may only have one per type') unless single_template?
end
```